samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,401 @@
+		break;
+	case LOCATION_EVT_GNSS_ASSISTANCE_REQUEST:
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
+		mosh_print("MoSh: A-GPS request from modem: emask:0x%08X amask:0x%08X flags:%d",
+		event_data->request.sv_mask_ephe,
+		event_data->request.sv_mask_alm,
+		event_data->request.data_flags);
+#endif
+		break;
+	}
+}
+
+void location_ctrl_init(void)
+{
+	int ret;
+
+	ret = location_init(location_ctrl_event_handler);
+	if (ret) {
+		mosh_error("Initializing the Location library failed, err: %d\n", ret);
+	}
+}
+
+/******************************************************************************/
+
+int location_shell(const struct shell *shell, size_t argc, char **argv)
+{
+	struct location_shell_cmd_args loc_cmd_args = {
+		.command = LOCATION_CMD_NONE
+	};
+
+	int interval = 0;
+	bool interval_set = false;
+
+	int gnss_timeout = 0;
+	bool gnss_timeout_set = false;
+
+	enum location_accuracy gnss_accuracy = 0;
+	bool gnss_accuracy_set = false;
+
+	int cellular_timeout = 0;
+	bool cellular_timeout_set = false;
+	enum location_service cellular_service = LOCATION_SERVICE_ANY;
+
+	int wifi_timeout = 0;
+	bool wifi_timeout_set = false;
+	enum location_service wifi_service = LOCATION_SERVICE_ANY;
+
+	enum location_method method_list[CONFIG_LOCATION_METHODS_LIST_SIZE] = { 0 };
+	int method_count = 0;
+
+	int opt;
+	int ret = 0;
+	int long_index = 0;
+
+	if (argc < 2) {
+		goto show_usage;
+	}
+
+	/* command = argv[0] = "location" */
+	/* sub-command = argv[1] */
+	if (strcmp(argv[1], "get") == 0) {
+		loc_cmd_args.command = LOCATION_CMD_GET;
+	} else if (strcmp(argv[1], "cancel") == 0) {
+		loc_cmd_args.command = LOCATION_CMD_CANCEL;
+	} else {
+		mosh_error("Unsupported command=%s\n", argv[1]);
+		ret = -EINVAL;
+		goto show_usage;
+	}
+
+	/* Reset getopt due to possible previous failures */
+	optreset = 1;
+
+	/* We start from subcmd arguments */
+	optind = 2;
+
+	while ((opt = getopt_long(argc, argv, "m:", long_options, &long_index)) != -1) {
+		switch (opt) {
+		case LOCATION_SHELL_OPT_GNSS_TIMEOUT:
+			gnss_timeout = atoi(optarg);
+			if (gnss_timeout == 0) {
+				mosh_error(
+					"GNSS timeout (%d) must be positive integer. ",
+					gnss_timeout);
+				return -EINVAL;
+			}
+			gnss_timeout_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_CELLULAR_TIMEOUT:
+			cellular_timeout = atoi(optarg);
+			if (cellular_timeout == 0) {
+				mosh_error(
+					"Cellular timeout (%d) must be positive integer. ",
+					cellular_timeout);
+				return -EINVAL;
+			}
+			cellular_timeout_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_CELLULAR_SERVICE:
+			cellular_service = location_shell_string_to_service(optarg);
+			if (cellular_service == MOSH_LOC_SERVICE_NONE) {
+				mosh_error("Unknown cellular positioning service. See usage:");
+				goto show_usage;
+			}
+			break;
+
+		case LOCATION_SHELL_OPT_WIFI_TIMEOUT:
+			wifi_timeout = atoi(optarg);
+			if (wifi_timeout == 0) {
+				mosh_error(
+					"WiFi timeout (%d) must be positive integer. ",
```

Comment:
"WiFi" should be changed to "Wi-Fi" everywhere in the PR (at least in all output and documentation).